packages/next-server/server/next-server.ts

```diff
@@ -272,7 +272,7 @@ export default class Server {
 
     const html = await this.renderToHTML(req, res, pathname, query, {
       amphtml: query.amp && this.nextConfig.experimental.amp,
-      dataOnly: req.headers && (req.headers.accept || '').indexOf('application/amp.bind+json') !== -1,
+      dataOnly: Boolean(query.dataOnly) || (req.headers && (req.headers.accept || '').indexOf('application/amp.bind+json') !== -1),
```

Comment:
I think we should rename the query 🔜 but for now it's fine.

Reply:
Also this check needs to check for the config value, otherwise it's going to be enabled when we don't want it to be.

Reply:
👌

Reply:
Updated